clientd3d/ownerdrw.c

```diff
@@ -374,6 +374,7 @@ void DrawOwnerListItem(const DRAWITEMSTRUCT *lpdis, Bool selected, Bool combo)
    SetBkMode(lpdis->hDC, TRANSPARENT);
 
    if (style & (OD_DRAWOBJ | OD_DRAWICON) && obj != NULL)
+      if (IsValidRarity(obj->rarity)) 
```

Comment:
I don't think this is necessary.  In C, a cast can't crash.

An access violation means referencing bad memory, which strongly suggests that obj is NULL.  Of course, there's a check immediately preceding this line, so that can't be the problem.  But it could be a problem here:

https://github.com/Meridian59/Meridian59/blob/bf65a1a5677e277379662456ca42f96106045b5b/clientd3d/ownerdrw.c#L461

I wonder if the compiler has drastically moved things around and done the dereferencing here in the compiled code.  It would be worth checking for nullptr before that line,

Reply:
Added a null pointer check right before the call to `GetPlayerNameColor`.  If the obj is null then we use default black text.